### ISSUE TITLE
protobuf-config.cmake: Invalid escape sequence \.

### ISSUE THREAD

**Issue by Croydon** [protocolbuffers/protobuf#7759]
**What version of protobuf and what language are you using?**
Version: 3.12.4
Language: C++

**What operating system (Linux, Windows, ...) and version?**  
Reproducable both on Windows 10 + WSL Fedora Remix 31 

**What runtime / compiler are you using (e.g., python version or gcc version)**
Happens both with CMake 3.17.3 on Windows and CMake 3.18.0 on WSL Fedora Remix 31

**What did you do?**
Try to use the `PROTOBUF_GENERATE_CPP` function

**What did you expect to see**
No CMake error at least 😄 

**What did you see instead?**
```
CMake Error at lib/cmake/protobuf/protobuf-config.cmake:108 (if):
  Syntax error in cmake code at
    lib/cmake/protobuf/protobuf-config.cmake:108

  when parsing string

    ^\.\.[/\\].*

  Invalid escape sequence \.
Call Stack (most recent call first):
  lib/cmake/protobuf/protobuf-module.cmake:22 (protobuf_generate)
  CMakeLists.txt:13 (PROTOBUF_GENERATE_CPP)
```

The respective code lines got recently introduced via

https://github.com/protocolbuffers/protobuf/commit/c91ab40081b1916b286c6c9a4ce7fce4c1125a87
https://github.com/protocolbuffers/protobuf/pull/7359

Therefore a friendly ping to @afshinpir and @acozzette 😄

**Comment by afshinpir** [comment]
Hi.

It seems the CMake version problem again. I made a mistake when updating code and used 2 features from CMake 3.14 incorrectly. I will update code immediately.

**Comment by afshinpir** [comment]
@Croydon I updated code to support previous version of CMake too and already made a pull request. But it takes few days to pass checks and merges.
It is #7761 .

**Comment by Croydon** [comment]
If I understand you correctly then you are assuming that the code only makes problems with _**older**_ CMake versions than 3.14, but in my case I have problems with this code with 3.17 and 3.18 🤔

**Comment by afshinpir** [comment]
@Croydon Yea, I made a mistake and used 2 newer features of CMake and I didn't noticed because I had newer versions of CMake myself and PR tests passed here too. 

Since you say that you have problems with CMake 3.17 and 3.18, I believe that it is because in my PC, I had `cmake_minimum_required(VERSION 3.14)` as minimum version of CMake for my project. Because the previous code works on my PC without any problem with CMake 3.16 and the only difference is this minimum version line.

But after updating this code in order to support all previous versions in this new PR, I changed my project to `cmake_minimum_required(VERSION 3.1)`  and it works for me, so I think everything will be solved.

If it is urgent, you can add changes from my PR (it is 5 lines of code) to your CMake until it merge into master.

**Comment by acozzette** [comment]
This should now be fixed thanks to #7761 from @afshinpir.
